### PR TITLE
GH-4 - start of checklist page

### PR DESCRIPTION
**Pull request by ericschoenborn**
simple page concept

### CODE REVIEW

**Comment by mwi-pharju** [comment]
It appears that prettier replaced all the `'`s with `"` in two of the files, but otherwise there aren't really any material code changes. I recommend avoiding mixing these with material code changes, otherwise it can be difficult to determine what was actually added.

File: src/ChecklistDojo/ClientApp/src/components/List.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      });
+  }
+
+  static MakeCheckList(listItems) {
+    return (
+      <ul>
+        {listItems.map(item =>
+          item.checked ? (
+            <li>
+              <input key={item.key} type="checkbox" checked /> {item.lable}{" "}
```
Comment (mwi-pharju) [inline src/ChecklistDojo/ClientApp/src/components/List.js]: `lable `

Comment (mwi-pharju) [reply src/ChecklistDojo/ClientApp/src/components/List.js]: These two components look exactly the same other than the inclusion of the checked property, you should be able to do this for simplicity and remove the ternary:
```
checked={item.checked}
```

File: src/ChecklistDojo/ClientApp/src/components/List.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      { key: 4, lable: "whats step 2?", checked: false }
+    ];
+    fetch("api/SampleData/WeatherForecasts")
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ listItems: fakeQueryData });
+      });
+  }
+
+  static MakeCheckList(listItems) {
```
Comment (mwi-pharju) [inline src/ChecklistDojo/ClientApp/src/components/List.js]: This method should be its own component.

File: src/ChecklistDojo/ClientApp/src/components/List.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    );
+  }
+
+  render() {
+    var contents = List.MakeCheckList(this.state.listItems);
+    console.log(this.props);
+    var title = "default";
+
+    if (this.props) {
+      console.log(this.props.match.params.title);
```
Comment (mwi-pharju) [inline src/ChecklistDojo/ClientApp/src/components/List.js]: I'd rather not commit `console.log`s

**Comment by mwi-pharju** [comment]
Can I see a screenshot of what this looks like?

File: src/ChecklistDojo/ClientApp/src/components/List.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    this.state = { listItems: [], loading: true };
+
+    var fakeQueryData = [
+      { key: 1, lable: "step one: steal underpants", checked: true },
+      { key: 2, lable: "step two: ", checked: false },
+      { key: 3, lable: "step three: profit", checked: true },
+      { key: 4, lable: "whats step 2?", checked: false }
+    ];
+    fetch("api/SampleData/WeatherForecasts")
```
[the remark this answers is not in the document]
Comment (ericschoenborn) [reply src/ChecklistDojo/ClientApp/src/components/List.js]: 9d59553a8bf843cb3bfa45587ecd6b6259434d38

File: src/ChecklistDojo/ClientApp/src/components/List.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+﻿import React, { Component } from "react";
+
+export class List extends Component {
```
[the remark this answers is not in the document]
Comment (ericschoenborn) [reply src/ChecklistDojo/ClientApp/src/components/List.js]: 7599026acb6bc8652f7d8813e09d1f55aa5a661e

File: src/ChecklistDojo/ClientApp/src/components/List.js (hunk 2 above)
Comment (ericschoenborn) [reply src/ChecklistDojo/ClientApp/src/components/List.js]: f09a5ff8ca8988fbaa1f44b8858b01b84f73fdaf

**Comment by ericschoenborn** [comment]
4paul

File: src/ChecklistDojo/ClientApp/src/components/List.js (hunk 2 above)
Comment (mwi-pharju) [reply src/ChecklistDojo/ClientApp/src/components/List.js]: It looks like you extracted the list of checklist items into its own component, what I wanted was to have the checklist item itself extracted into its own component, which would look something like this:
```
// ChecklistItem,jsx
export default (props) =>  {
  const {key, checked, text} = props;
  return (
    <input id={key} type="checkbox" checked={checked} /> {text}
  );
}
```
I'm starting pull back on wanting it to be called `InstanceItem`. `InstanceItem` or `ChecklistItem` would both be ok for the time being, I think.

Comment (mwi-pharju) [reply src/ChecklistDojo/ClientApp/src/components/List.js]: I have made the relevant change 9f3476d

**Pull request merged**